### DISCUSSION TITLE
connecting to thorium browser

### DISCUSSION THREAD

**Discussion by adelpourmand** [keepassxreboot/keepassxc#11349] [Q&A] [answered]
Hello, I can't connect keepassxc to a chromium-based browser called Thorium (https://thorium.rocks/). Is there any way to do so?

Thank you in advance . . .
P.S.:
KeepassXC version 2.7.9
OS: Linux mint 22

**Comment by droidmonkey** [comment, marked as answer]
Use the custom browser feature in the browser settings, advanced tab.

**Reply by adelpourmand** [reply]
Thank you for your reply, I don't know why but that feature which you mentioned does not work. keepassxc says: "Could not save the native messaging script file for custom" . For this reason I followed reading below webpage:
"https://github.com/keepassxreboot/keepassxc-browser/wiki/Troubleshooting-guide", 
Then I wrote a json file with bellow text in it:
{
    "allowed_origins": [
        "chrome-extension://pdffhmdngciaglkoonimfcmckehcpafo/",
        "chrome-extension://oboonakemofpalcgghocfoadofidjkkk/"
    ],
    "description": "KeePassXC integration with native messaging support",
    "name": "org.keepassxc.keepassxc_browser",
    "path": "/usr/bin/keepassxc-proxy",
    "type": "stdio"
}
Afterwards, i copied the json file to ~/.config/thorium/NativeMessagingHosts, and it works now!
Just, please give me a feedback and let me know if I did it in a right and secure way

Thank you.

**Reply by droidmonkey** [reply]
That works, so you tried putting `~/.config/thorium/NativeMessagingHosts` into the custom browser setting? If that failed then it is likely a permissions issue

**Reply by adelpourmand** [reply]
Mmm.... you're right. As you said the problem of "Could not save the native messaging script file for custom" is related to permission. To address it i check-marked chromium in general tab. Then i check-marked custom browser in advanced tab and selected browser type as chromium. Afterwards i browsed to following address for config location:
~/.config/chromium/NativeMessagingHosts
